core/src/test/java/org/apache/iceberg/LocalTableOperations.java

```diff
@@ -63,7 +65,9 @@ public String metadataFileLocation(String fileName) {
         fileName,
         name -> {
           try {
-            return temp.newFile(name).getAbsolutePath();
+            return java.nio.file.Files.createTempDirectory(temp, "junit")
```

Comment:
shouldn't this create a temp file with the given name rather than a temp directory?